python/orchestrator/orchestrator.py

```diff
@@ -0,0 +1,307 @@
+            return Response.OK
+        else:
+            self.__logger.critical(
+                f'Global state must be {valid_state} for executing'
+                f'the steering command: {steering_command}')
+            return Response.ERROR
+
+    def __set_up(self):
+        """
+        helper function for setting up before orchestration.
+        """
+        # register with registry
+        self.__component_service_registry_manager.register(
+                        os.getpid(),  # id
+                        SERVICE_COMPONENT_CATEGORY.ORCHESTRATOR,   # category
+                        SERVICE_COMPONENT_CATEGORY.ORCHESTRATOR,   # name
+                        (self.__orchestrator_in_queue,  # endpoint
+                         self.__orchestrator_out_queue),
+                        SERVICE_COMPONENT_STATUS.UP,  # current status
+                        STATES.READY)  # current state
+        # retreive registered component which is later needed to update states
+        self.__orchestrator_registered_component =\
+            self.__component_service_registry_manager.find_by_id(os.getpid())
+        if(self.__orchestrator_registered_component is not None):
+            # set the flag to indicate successful registration
+            self.__logger.debug(
+                f'component service id: '
+                f'{self.__orchestrator_registered_component.id}'
+                f'; name: {self.__orchestrator_registered_component.name}')
+            self.__is_registered.set()
+        else:
+            # not found in registry
+            self.__logger.critical('quitting! not found in registry.')
+            signal.raise_signal(signal.SIGTERM)
+            return Response.ERROR
+        # fetch C&S from regitstry
+        self.__command_and_steering_service =\
+            self.__get_component_from_registry(
+                        SERVICE_COMPONENT_CATEGORY.COMMAND_AND_SERVICE)
+        self.__logger.debug(f'command and steering service: '
+                            f'{self.__command_and_steering_service[0]}')
+        # fetch C&S endpoint (in_queue and out_queue)
+        self.__command_and_steering_service_in_queue,\
+            self.__command_and_steering_service_out_queue =\
+            self.__command_and_steering_service[0].endpoint
+        # initialize the Communicator object for communication via Queues
+        self.__communicator = CommunicatorQueue(self._log_settings,
+                                                self._configurations_manager)
+        # update global state to READY assuming all components are already
+        # launched by the launcher successfully. The local states are anyway
+        # validated during the update process.
+        self.__health_status_keeper.update_global_state()
+        # start monitoring threads
+        self.__health_status_keeper.start_monitoring()
+        self.__alarm_signal_monitor.start_monitoring()
+        return Response.OK
+
+    def run(self):
```

Comment:
same as above in the AppCompanion and the Steering Service:
The function called bz the launcher is start().
See comment in launcher.py.

Reply:
Orchestrator, Application_Companions and Steering Service are separate processes. We override run() method of multiprocessing to implement the functionality to be executed by a new process. However, it is not recommended to call it explicitly rather we call start() to arrange for the run() method to be invoked in a separate process. For more details: https://docs.python.org/3/library/multiprocessing.html